node/lib/util/submodule_config_util.js

```diff
@@ -271,34 +273,35 @@ exports.initSubmodule = co.wrap(function *(repoPath, name, url) {
 });
 
 /**
- * Open the submodule having the specified `name` and `url` for the repo at the
- * specified `repoPath`.  Configure the repository for this submodule to have
- * `url` as its remote, unless `url` is relative, in which case resolve `url`
- * against the specified `repoUrl`.  Throw a `UserError` if
- * `null === repoUrl` and `url` is relative.  Return the newly opened
- * repository.  Note that this command does not fetch any refs from the remote
- * for this submodule, and while its repo can be opened it will be empty.
+ * Open the submodule having the specified `name` and `url` for the `metaRepo`.
+ * Configure the repository for this submodule to have `url` as its remote,
+ * unless `url` is relative, in which case resolve `url` against the specified
+ * `repoUrl`.  Throw a `UserError` if `null === repoUrl` and `url` is relative.
```

Comment:
I don't think it asserts in that situation. It will assert if `repoUrl` is non-null and non-string.